metadata-ingestion/tests/unit/utilities/test_yaml_sync_utils.py

```diff
@@ -0,0 +1,91 @@
+import pathlib
+
+from datahub.utilities.yaml_sync_utils import YamlFileUpdater
+
+
+def test_update_yaml_file(tmp_path: pathlib.Path) -> None:
+    infile = tmp_path / "test.yml"
+
+    # Note - this will drop the leading newline before the comment.
+    infile.write_text(
+        """
+# this is a comment
+#
+obj:
+  key1: value1
+
+list_ty:
+  - foo
+  - key1: value1
+    key2: value2
+"""
+    )
+    # ind=4, bsi=2
+
+    with YamlFileUpdater(infile) as doc:
+        doc["foo"] = "bar"
+        doc["list_ty"].append("baz")
+        doc["list_ty"][1]["key1.5"] = "val1.5"
```

Comment:
Can you test replacing data too? This looks like it's only adding new key-values or list entries

Reply:
the real purpose of this test is checking that the comments are retained - the changes themselves are just dummy changes